Model/Order/PlaceOrder.php

```diff
@@ -137,11 +181,234 @@ public function place(string $shopId, OrderDataInterface $order): ResultInterfac
+             *     tender_type: string,
+             *     tender_details: array{
+             *         brand: string,
+             *         last_four: string,
+             *         bin: string,
+             *         expiration: string
+             *     },
+             *     gateway_response_data: string[]
+             * } $transaction
+             *
+             * TODO: verify that the status used for filtering is correct
+             */
+            static fn (array $transaction): bool => $transaction['status'] !== 'failure'
+        );
+
+        if (count($transactions) === 0) {
+            return $this->responseFactory->create(
+                [
+                    'errors' => [
+                        $this->errorFactory->create(
+                            [
+                                'message' => __(
+                                    'There are no successful transactions available for order "%1"',
+                                    $publicOrderId
+                                ),
+                                'code' => 500,
+                                'type' => 'server.bold_checkout_api_error'
+                            ]
+                        )
+                    ]
+                ]
+            );
+        }
+
+        /**
+         * @var array{
+         *     gateway: string,
+         *     payment_id: string,
+         *     amount: int,
+         *     transaction_id: string,
+         *     currency: string,
+         *     step: string,
+         *     status: 'success'|'',
+         *     tender_type: string,
+         *     tender_details: array{
+         *         brand: string,
+         *         last_four: string,
+         *         bin: string,
+         *         expiration: string
+         *     },
+         *     gateway_response_data: string[]
+         * } $firstTransaction
+         */
+        $firstTransaction = array_shift($transactions);
+        /** @var OrderPaymentInterface $orderPayment */
+        $orderPayment = $this->paymentFactory->create();
+        /** @var TransactionInterface $transaction */
+        $transaction = $this->transactionFactory->create();
+        /** @var OrderDataInterface $orderData */
+        $orderData = $this->orderDataFactory->create();
+        [$cardExpirationMonth, $cardExpirationYear] = explode(
+            '/',
+            $firstTransaction['tender_details']['expiration'],
+            2
+        );
+
+        $orderPayment->setBaseAmountPaid($firstTransaction['amount'] / 100);
+        $orderPayment->setAmountPaid($firstTransaction['amount'] / 100);
+        $orderPayment->setCcLast4($firstTransaction['tender_details']['last_four']);
+        $orderPayment->setCcType($firstTransaction['tender_details']['brand']);
+        $orderPayment->setCcExpMonth($cardExpirationMonth);
+        $orderPayment->setCcExpYear($cardExpirationYear);
+
+        $transaction->setTxnId($firstTransaction['transaction_id']);
```

Comment:
[NIT] I think we can extract these into separate functions to handle orderPayment, and transactions.